lib/lolcommits/vcs_info.rb

```diff
@@ -0,0 +1,49 @@
+# -*- encoding : utf-8 -*-
+module Lolcommits
+  # base class ala plugin.rb
+  class VCSInfo
+    include Methadone::CLILogging
+    attr_accessor :sha, :message, :repo_internal_path, :repo, :url,
+                  :author_name, :author_email, :branch
+
+    def self.repo_root?(path = '.')
+      GitInfo.repo_root?(path) || MercurialInfo.repo_root?(path)
+    end
+
+    def initialize
+      puts 'Base VCS, no implementation'
```

Comment:
I think raising an error is a better approach e.g.

    def message
      raise NotImplementedError.new("#{self.class.name} is a base class, implement this method in a subclass")
    end